src/Components/Article/ArticleInfo.jsx

```diff
@@ -2,16 +2,45 @@ import { useEffect, useState } from "react";
 import { useParams } from "react-router-dom";
 import CommentCard from "./CommentCard";
 import { format } from "date-fns";
-import { getArticleByID, getCommentsByID } from "../../../utils";
+import {
+  getArticleByID,
+  getCommentsByID,
+  patchArticleVotes,
+} from "../../../utils";
 
 export default function ArticleInfo() {
   const { article_id } = useParams();
   const [currArticle, setCurrArticle] = useState({});
   const [comments, setComments] = useState([]);
   const [loading, setLoading] = useState(true);
   const [error, setError] = useState(null);
+  const [upvote, setUpvote] = useState(false);
+  const [downvote, setDownvote] = useState(false);
+  const [commentError, setCommentError] = useState(null);
 
-  console.log(currArticle);
+  const handleUpvote = () => {
+    //votechange will be 1 if neither buttons are toggle, 2 if downvote is already toggled or -1 if i untoggle upvote. this is then passed into my api for patch method.
+    const voteChange = upvote ? -1 : downvote ? 2 : 1;
+    setUpvote(!upvote);
+    setDownvote(false);
+    setCurrArticle({ ...currArticle, votes: currArticle.votes + voteChange });
+
+    patchArticleVotes(article_id, voteChange).catch((err) => {
+      setCurrArticle({ ...currArticle, votes: currArticle.votes - voteChange });
+      setCommentError("Could not update votes, try again later.");
+    });
+  };
+
+  const handleDownvote = () => {
+    const voteChange = downvote ? 1 : upvote ? -2 : -1;
+    setDownvote(!downvote);
+    setUpvote(false);
+    setCurrArticle({ ...currArticle, votes: currArticle.votes + voteChange });
+    patchArticleVotes(article_id, voteChange).catch((err) => {
+      setCurrArticle({ ...currArticle, votes: currArticle.votes - voteChange });
+      setCommentError("Could not update votes, try again later.");
+    });
```

Comment:
Your logic for when the vote fails works really well. But once the error is displayed, it stays on the page even if the vote does work when pressed again. Please do ensure this message disappears if the vote is successful